src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -85,7 +87,11 @@ public final void runLocked(final Runnable function) {
                 lock = new InterProcessSemaphoreMutex(curatorFramework, "/nakadi/locks/subscription_" + subscriptionId);
             }
 
-            lock.acquire();
+            final boolean acquired = lock.acquire(SECONDS_TO_WAIT_FOR_LOCK, TimeUnit.SECONDS);
+            if (!acquired) {
+                throw new ServiceUnavailableException("Failed to acquire subscription lock within " +
```

Comment:
maybe also log some specific log message here?

Reply:
I use 2 techniques, that help me a lot: 
1. throw Exception
2. log message
I try to never mix it. And here is the same. Log message will be logged in place, where exceptions are logged.

Reply:
Oh yes, sorry. I forgot that it will be caught and logged anyway.